### ISSUE TITLE
TensorFlow 1.0

### ISSUE THREAD

**Issue by tomkocmi** [ufal/neuralmonkey#341]
TensorFlow 1.0 is no longer release candidate but it is released. Shouldn't we update the NM to this version as soon as possible?

**Comment by jindrahelcl** [comment]
There are issues with transferring to a newer of tensorflow. Discussed mostly in #156. You can try and fix that. There will also be a lot of deprecated stuff that should be updated.

Before we start working on this... Měli bychom se zamyslet, jestli to ukládání proměnných funguje tak, jak chceme, a jestli to nechceme nějak překopat, protože k tomu máme jedinečnou příležitost - pokud se to teď "zamázne" nějakym hotfixem, tak se k tomu už nikdy v životě nedostanem, protože to nebude mít takovou prioritu. @jlibovicky @tomasmcz

**Comment by jindrahelcl** [comment]
(btw. linking to [developers' checklist](https://github.com/ufal/neuralmonkey/wiki/Developers'-checklists))

**Comment by tomkocmi** [comment]
Dival jsem se, jestli je neco fakt zajimaveho v TF 1.0 a nasel jsem toto, tak jen takova drobna motivace proc prejit na TF 1.0 ;)

> It’s faster: TensorFlow 1.0 is incredibly fast! XLA lays the groundwork for even more performance improvements in the future, and tensorflow.org now includes tips & tricks for tuning your models to achieve maximum speed. We’ll soon publish updated implementations of several popular models to show how to take full advantage of TensorFlow 1.0 – including a 7.3x speedup on 8 GPUs for Inception v3 and 58x speedup for distributed Inception v3 training on 64 GPUs!
> 
> It’s more flexible: TensorFlow 1.0 introduces a high-level API for TensorFlow, with tf.layers, tf.metrics, and tf.losses modules. We’ve also announced the inclusion of a new tf.keras module that provides full compatibility with Keras, another popular high-level neural networks library.
> 
> It’s more production-ready than ever: TensorFlow 1.0 promises Python API stability (details here), making it easier to pick up new features without worrying about breaking your existing code.

Jinak zde je popsano, jake zmeny je potreba ohlidat pri prechodu na TF 1.0
https://www.tensorflow.org/install/migration
